### PR TITLE
G-10122 Fix issues related to geometry buffering

### PR DESCRIPTION
**Pull request by jrnorth**
#### What does this PR do?
- `SolrFilterDelegate` will remove holes from polygons and multipolygons that cross the antimeridian. When handling polygons that cross the antimeridian, spatial4j requires any interior rings to be subsets of the exterior ring, which seems wrong (see https://github.com/locationtech/spatial4j/blob/2926812ae302c6e0f24fb5995b1fd24be7fe0b56/src/main/java/org/locationtech/spatial4j/shape/jts/JtsGeometry.java#L482). Removing any holes means this check won't fail.
- Enables `normWrapLongitude` in `SolrFilterDelegate`'s `SpatialContext`. If a shape near the antimeridian is buffered, then it may have points that end up outside the range [-180, 180]. Enabling this allows spatial4j to handle those points correctly.
- Enables `allowMultiOverlap` in `SolrFilterDelegate`'s `SpatialContext`. Sometimes, we have polygons that share a border, such as the polygons of the Russia WKT on the antimeridian:
Screen Shot 2022-05-23 at 1 29 47 PM
Adding a buffer causes them to overlap, so enabling this allows spatial4j to handle those points correctly.

#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@jlcsmith
@kcwire 
@millerw8

#### How should this be tested?
1. Run `gazetteer:update` with https://github.com/codice/ddf/blob/03be912e331024f3bc6856d5e87f8e3b7db0b392/distribution/ddf-common/src/main/resources/data/countries.geo.json
2. Run `gazetteer:build-suggester-index`
3. Ingest the following files:
`russia1.json`
```
{
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [-179.9375, 69.116389]
  },
  "properties": {
    "title": "Russia 1"
  }
}
```
`russia2.json`
```
{
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [179.900833, 71.637778]
  },
  "properties": {
    "title": "Russia 2"
  }
}
```
These locations are both within a 25km buffer of the Russia polygon, close to the antimeridian.
4. Open Intrigue and create an anyGeo query with the keyword 'Russia'. Run it and verify you get no results.
5. Add a buffer of 25km and verify you get both results.

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Review by jlcsmith — APPROVED** [review]

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java [hunk 1]
```diff
@@ -723,6 +758,18 @@ public SolrQuery dwithin(String propertyName, String wkt, double distance) {
...
+        final boolean crossesDateline =
+            envelope.getWidth() >= 180
+                || (envelope.getMinX() <= 180 && envelope.getMaxX() > 180)
+                || (envelope.getMinX() < -180 && envelope.getMaxX() >= -180);
+        // When spatial4j's JtsGeometry unwraps polygons that cross the dateline, it checks that all
+        // interior rings of the polygon are subsets of the exterior ring, which means polygons with
+        // holes will throw an exception. Remove any holes from polygons that cross the dateline to
+        // prevent that exception from being thrown.
+        if (crossesDateline) {
+          bufferGeo = removeHoles(bufferGeo);
```
Comment (pklinef) [inline catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java:771]: Looks like there is a generic hole polygon hole remover but it is in the jtslab package so it is not production ready yet.

https://github.com/locationtech/jts/blob/0afbfb1956ec24912a8b4dc4edff0f1200442857/modules/lab/src/main/java/org/locationtech/jtslab/clean/HoleRemover.java

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java [hunk 2]
```diff
@@ -709,6 +717,33 @@ public SolrQuery contains(String propertyName, String wkt) {
     return operationToQuery("Contains", propertyName, wkt);
   }
 
+  public static Geometry removeHoles(final Geometry geo) {
```
Comment (pklinef) [inline catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java:720]: ✏️ We should probably log something at a lower level when we remove holes from the given geo.

File: catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/SolrFilterDelegateTest.java [hunk 3]
```diff
@@ -141,17 +145,19 @@ public void polygonWithHoleAndSelfIntersecting() {
...
+  public void multiPolygon() throws ParseException {
     String wkt =
         "MULTIPOLYGON (((30 20, 45 40, 10 40, 30 20)), ((15 5, 40 10, 10 20, 5 10, 15 5)))";
     when(mockResolver.getField(
             "testProperty", AttributeFormat.GEOMETRY, false, Collections.emptyMap()))
         .thenReturn("testProperty_geohash_index");
     SolrQuery query = toTest.contains("testProperty", wkt);
+    MultiPolygon multiPolygon = (MultiPolygon) new WKTReader().read(wkt);
+    // Need to use union since allowMultiOverlap is enabled
+    MultiPolygon unioned = (MultiPolygon) new UnaryUnionOp(multiPolygon).union();
```
Comment (pklinef) [inline catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/SolrFilterDelegateTest.java:157]: ❓ Is this new union requirement going to impact geos read in from other bundles in the system?

**Review by pklinef — APPROVED** [review]

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java (hunk 1 above)
Comment (jrnorth) [reply catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java:771]: Good find; I will add a comment saying we should look into switching to this once it's ready.

File: catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/SolrFilterDelegateTest.java (hunk 3 above)
Comment (jrnorth) [reply catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/SolrFilterDelegateTest.java:157]: Maybe I need to reword this comment. Nothing new is required of WKTs being read by this bundle. When `allowMultiOverlap` is enabled, spatial4j will automatically compute the union of any `GeometryCollection` instance. This comment is just saying that we need to compute the union of the expected WKT for the assertion.

**Comment by jrnorth** [comment]
build now

**Comment by jrnorth** [comment]
build now

**Comment by jrnorth** [comment]
build now

**Comment by jlcsmith** [comment]
Verified -
- Configured solr schema with normWrapLongitude="true"
- ingested russia1/russia2 json metacards
- Using karaf CLI, verified a query using the Russia geo with a 2 km buffer **did not** find the 2 metacards 
``` catalog:search  --cql "DWITHIN(anyGeo, MULTIPOLYGON (((106.9701 76.9742, 107.24 76.48, 111.0773 76.71, 114.1342 75.8476, 113.8854 75.3278, 109.4 74.18, 112.1192 73.7877, 113.0195 73.9769, 113.5296 73.3351, 115.5678 73.7529, 118.7763 73.5877, 119.02 73.12, 123.2007 72.9712, 123.2578 73.735, 126.9764 73.5655, 128.5913 73.0387, 129.0516 72.3987, 128.46 71.98, 129.716 71.193, 131.2886 70.787, 132.2535 71.8363, 133.8577 71.3864, 135.5619 71.6553, 137.4976 71.3476, 138.2341 71.628, 139.8698 71.4878, 139.1479 72.4162, 140.4682 72.8494, 149.5 72.2, 150.3512 71.6064, 152.9689 70.8422, 158.9978 70.8667, 159.8303 70.4532, 159.7087 69.722, 160.9405 69.4373, 167.8357 69.5827, 169.5776 68.6938, 170.8169 69.0136, 170.0082 69.6528, 170.4535 70.097, 175.724 69.8773, 180 68.9636, 180 64.9797, 178.7072 64.5349, 177.4113 64.6082, 179.3703 62.9826, 179.2283 62.3041, 177.3643 62.5219, 173.6801 61.6526, 170.6985 60.3362, 170.3309 59.8818, 168.9005 60.5736, 166.295 59.7886, 165.84 60.16, 164.8767 59.7316, 163.5393 59.8687, 163.2171 59.211, 162.0173 58.2433, 162.053 57.8391, 163.1919 57.615, 163.0579 56.1592, 162.1296 56.1222, 161.7015 55.2857, 162.1175 54.8551, 160.3688 54.3443, 160.0217 53.2026, 158.5309 52.9587, 158.2312 51.9427, 156.7898 51.0111, 155.4337 55.381, 155.9144 56.7679, 156.7582 57.3647, 156.8104 57.832, 158.3643 58.0558, 161.872 60.343, 163.6697 61.1409, 164.4736 62.5506, 163.2584 62.4663, 162.6579 61.6425, 160.1215 60.5442, 159.3023 61.774, 156.7207 61.4344, 154.2181 59.7582, 155.0438 59.145, 151.2657 58.7809, 151.3382 59.504, 149.7837 59.6557, 148.5448 59.1645, 145.4872 59.3364, 142.1978 59.04, 135.1262 54.7296, 136.7017 54.6036, 137.1934 53.9773, 138.1647 53.755, 138.8046 54.2546, 139.9015 54.1897, 141.3453 53.0896, 141.3792 52.2388, 140.5974 51.2397, 140.0619 48.4467, 138.2197 46.308, 134.8694 43.3982, 133.5369 42.8115, 132.9063 42.7985, 132.2781 43.2846, 130.78 42.22, 130.6339 42.903, 131.1447 42.93, 131.2886 44.1115, 131.0252 44.968, 131.8835 45.3212, 133.0971 45.1441, 135.0263 48.4782, 132.5067 47.789, 130.9873 47.7901, 130.5823 48.7297, 129.3978 49.4406, 127.6574 49.7603, 125.9463 52.7928, 125.0682 53.161, 123.5715 53.4588, 121.0031 53.2514, 120.1771 52.7539, 120.7258 52.5162, 120.7382 51.9641, 119.2794 50.5829, 119.2885 50.1429, 117.8792 49.511, 116.6788 49.8885, 115.4857 49.8052, 114.3625 50.2483, 112.8977 49.5436, 110.662 49.1301, 108.4752 49.2825, 106.8888 50.2743, 103.6765 50.09, 102.2559 50.5106, 102.0652 51.2599, 98.8615 52.0474, 97.8257 51.011, 98.2318 50.4224, 97.2598 49.7261, 94.8159 50.0134, 94.1476 50.4805, 92.2347 50.8022, 88.8056 49.4705, 87.36 49.215, 86.8294 49.8267, 85.5413 49.6929, 83.383 51.0692, 81.946 50.8122, 80.5684 51.3883, 80.0356 50.8648, 77.8009 53.4044, 76.5252 54.177, 76.8911 54.4905, 74.3848 53.5469, 73.4257 53.4898, 73.5085 54.0356, 72.2242 54.3767, 71.1801 54.1333, 70.8653 55.1697, 69.0682 55.3853, 65.1785 54.3542, 61.4366 54.0063, 60.9781 53.665, 61.7 52.98, 60.74 52.72, 60.9273 52.4475, 59.9675 51.9604, 61.588 51.2727, 61.3374 50.7991, 59.9328 50.8422, 59.6423 50.5454, 58.3633 51.0636, 56.778 51.0436, 55.7169 50.6217, 52.3287 51.7187, 50.7666 51.6928, 48.7024 50.6051, 48.5778 49.8748, 47.5495 50.4547, 46.4664 48.3942, 47.3152 47.7159, 48.0573 47.7438, 48.6947 47.0756, 48.5933 46.561, 49.1012 46.3993, 48.6454 45.8063, 47.6759 45.6415, 46.682 44.6092, 47.5909 43.6602, 47.4925 42.9866, 48.5844 41.8089, 47.8157 41.1514, 45.7764 42.0924, 45.4703 42.5028, 43.9312 42.555, 42.3944 43.2203, 39.955 43.435, 36.6755 45.2447, 37.4032 45.4045, 38.233 46.2409, 37.6737 46.6366, 39.1477 47.0448, 39.1212 47.2634, 38.2235 47.1022, 38.2551 47.5464, 39.7383 47.8989, 39.6747 48.7838, 40.069 49.6011, 38.0106 49.9157, 37.3935 50.384, 36.6262 50.2256, 35.3561 50.5772, 35.0222 51.2076, 34.2248 51.256, 34.3917 51.7689, 33.7527 52.3351, 31.786 52.1017, 31.3052 53.074, 32.6936 53.3514, 31.7313 53.794, 30.7575 54.8118, 30.8739 55.551, 29.3716 55.6701, 28.1767 56.1691, 27.77 57.2443, 27.2882 57.4745, 27.7167 57.7919, 27.4202 58.7246, 28.1317 59.3008, 27.9811 59.4754, 29.1177 60.0281, 28.07 60.5035, 31.14 62.3577, 31.5161 62.8677, 30.0359 63.5528, 30.4447 64.2045, 29.5444 64.9487, 30.2177 65.806, 29.0546 66.9443, 29.9774 67.6983, 28.4459 68.3646, 28.5919 69.0648, 32.1327 69.906, 33.7755 69.3014, 36.514 69.0634, 41.0599 67.4571, 41.126 66.7916, 38.383 65.9995, 33.9187 66.7596, 33.1844 66.6325, 34.8148 65.9002, 34.9439 64.4144, 37.0127 63.8498, 37.142 64.3347, 36.5396 64.7645, 37.176 65.1432, 39.5935 64.5208, 40.4356 64.7645, 39.7626 65.4968, 42.0931 66.4762, 43.9498 66.0691, 44.5323 66.7563, 43.6984 67.3525, 44.188 67.9505, 43.4528 68.5708, 46.25 68.25, 46.8213 67.69, 45.5552 67.5665, 45.562 67.0101, 46.3492 66.6677, 47.8942 66.8846, 48.1388 67.5224, 53.7174 68.8574, 54.4717 68.8082, 53.4858 68.2013, 54.7263 68.097, 55.4427 68.4387, 57.317 68.4663, 58.802 68.8808, 59.9414 68.2784, 61.0778 68.9407, 60.03 69.52, 60.55 69.85, 63.504 69.5474, 68.5122 68.0923, 69.1807 68.6156, 68.1352 69.3565, 66.9301 69.4546, 67.2598 69.9287, 66.6947 71.029, 68.5401 71.9345, 69.1964 72.8434, 69.94 73.04, 72.5875 72.7763, 72.796 72.2201, 71.8481 71.409, 72.4701 71.0902, 72.7919 70.3911, 72.5647 69.0209, 73.6679 68.4079, 73.2387 67.7404, 71.28 66.32, 72.423 66.1727, 73.921 66.7895, 74.1865 67.2843, 75.052 67.7605, 74.4693 68.329, 74.9358 68.9892, 73.8424 69.0715, 73.6019 69.6276, 74.3998 70.6318, 73.1011 71.4472, 74.8908 72.1212, 74.6593 72.8323, 75.158 72.855, 75.6835 72.3006, 75.289 71.3356, 76.3591 71.1529, 75.9031 71.874, 77.5767 72.2672, 79.652 72.3201, 81.5 71.75, 80.6107 72.5828, 80.5111 73.6482, 86.8223 73.9369, 86.0096 74.4597, 87.1668 75.1164, 92.9006 75.7733, 93.2342 76.0472, 96.6782 75.9155, 98.9225 76.4469, 100.7597 76.4303, 101.9908 77.2875, 104.3516 77.6979, 106.0666 77.3739, 104.705 77.1274, 106.9701 76.9742)), ((180 70.8322, 178.9034 70.7811, 178.7253 71.0988, 180 71.5157, 180 70.8322)), ((150.7317 75.0841, 149.5759 74.6889, 146.1192 75.173, 146.3585 75.4968, 150.7317 75.0841)), ((145.0863 75.5626, 144.3 74.82, 138.9554 74.6115, 136.9744 75.2617, 137.5118 75.9492, 138.8311 76.1368, 145.0863 75.5626)), ((143.648 50.7476, 144.6541 48.9764, 143.1739 49.3066, 142.5587 47.8616, 143.5335 46.8367, 143.5053 46.1379, 142.7477 46.7408, 142.092 45.9668, 141.9044 48.8592, 142.18 50.9523, 141.5941 51.9354, 141.6825 53.302, 142.6069 53.7621, 142.2097 54.2255, 142.6548 54.3659, 143.648 50.7476)), ((143.6039 73.2124, 139.8631 73.3698, 142.0621 73.8576, 143.6039 73.2124)), ((105.0755 78.3069, 99.4381 77.921, 101.2649 79.234, 102.0864 79.3464, 105.3724 78.7133, 105.0755 78.3069)), ((99.9398 78.8809, 94.9726 79.0447, 93.3129 79.4265, 92.5454 80.1438, 91.1811 80.3415, 95.9409 81.2504, 100.1867 79.7801, 99.9398 78.8809)), ((57.5357 70.7205, 53.6774 70.7627, 53.412 71.2067, 51.6019 71.4748, 51.4558 72.0149, 52.4783 72.2294, 52.4442 72.7747, 54.4276 73.6275, 53.5083 73.7498, 55.9025 74.6275, 55.6319 75.0814, 61.17 76.2519, 68.1571 76.9397, 68.8522 76.5448, 61.5835 75.2609, 58.4771 74.3091, 55.4193 72.3713, 55.6228 71.5406, 57.5357 70.7205)), ((51.1362 80.5473, 47.5861 80.0102, 46.5028 80.2472, 47.0725 80.5594, 44.847 80.5898, 48.3185 80.784, 48.5228 80.5146, 50.0398 80.9189, 51.5229 80.6997, 51.1362 80.5473)), ((22.7311 54.3275, 19.6606 54.4261, 19.8885 54.8662, 21.2684 55.1905, 22.7578 54.8566, 22.7311 54.3275)), ((-175.0142 66.5844, -174.3398 66.3356, -174.5718 67.0622, -171.8573 66.9131, -169.8996 65.9772, -170.8911 65.5414, -172.5302 65.4379, -172.555 64.4608, -172.9553 64.2527, -175.9835 64.9229, -176.2072 65.3567, -178.3599 65.3905, -178.9033 65.7404, -178.6861 66.1121, -179.8838 65.8746, -179.4327 65.4041, -180 64.9797, -180 68.9636, -174.9282 67.2059, -175.0142 66.5844)), ((-178.6938 70.893, -180 70.8322, -180 71.5157, -177.5779 71.2695, -178.6938 70.893))), 250,kilometers)"```
- Using karaf CLI, verified a query using the Russia geo with a 25 km buffer **did** find the 2 metacards 
``` catalog:search  --cql "DWITHIN(anyGeo, MULTIPOLYGON (((106.9701 76.9742, 107.24 76.48, 111.0773 76.71, 114.1342 75.8476, 113.8854 75.3278, 109.4 74.18, 112.1192 73.7877, 113.0195 73.9769, 113.5296 73.3351, 115.5678 73.7529, 118.7763 73.5877, 119.02 73.12, 123.2007 72.9712, 123.2578 73.735, 126.9764 73.5655, 128.5913 73.0387, 129.0516 72.3987, 128.46 71.98, 129.716 71.193, 131.2886 70.787, 132.2535 71.8363, 133.8577 71.3864, 135.5619 71.6553, 137.4976 71.3476, 138.2341 71.628, 139.8698 71.4878, 139.1479 72.4162, 140.4682 72.8494, 149.5 72.2, 150.3512 71.6064, 152.9689 70.8422, 158.9978 70.8667, 159.8303 70.4532, 159.7087 69.722, 160.9405 69.4373, 167.8357 69.5827, 169.5776 68.6938, 170.8169 69.0136, 170.0082 69.6528, 170.4535 70.097, 175.724 69.8773, 180 68.9636, 180 64.9797, 178.7072 64.5349, 177.4113 64.6082, 179.3703 62.9826, 179.2283 62.3041, 177.3643 62.5219, 173.6801 61.6526, 170.6985 60.3362, 170.3309 59.8818, 168.9005 60.5736, 166.295 59.7886, 165.84 60.16, 164.8767 59.7316, 163.5393 59.8687, 163.2171 59.211, 162.0173 58.2433, 162.053 57.8391, 163.1919 57.615, 163.0579 56.1592, 162.1296 56.1222, 161.7015 55.2857, 162.1175 54.8551, 160.3688 54.3443, 160.0217 53.2026, 158.5309 52.9587, 158.2312 51.9427, 156.7898 51.0111, 155.4337 55.381, 155.9144 56.7679, 156.7582 57.3647, 156.8104 57.832, 158.3643 58.0558, 161.872 60.343, 163.6697 61.1409, 164.4736 62.5506, 163.2584 62.4663, 162.6579 61.6425, 160.1215 60.5442, 159.3023 61.774, 156.7207 61.4344, 154.2181 59.7582, 155.0438 59.145, 151.2657 58.7809, 151.3382 59.504, 149.7837 59.6557, 148.5448 59.1645, 145.4872 59.3364, 142.1978 59.04, 135.1262 54.7296, 136.7017 54.6036, 137.1934 53.9773, 138.1647 53.755, 138.8046 54.2546, 139.9015 54.1897, 141.3453 53.0896, 141.3792 52.2388, 140.5974 51.2397, 140.0619 48.4467, 138.2197 46.308, 134.8694 43.3982, 133.5369 42.8115, 132.9063 42.7985, 132.2781 43.2846, 130.78 42.22, 130.6339 42.903, 131.1447 42.93, 131.2886 44.1115, 131.0252 44.968, 131.8835 45.3212, 133.0971 45.1441, 135.0263 48.4782, 132.5067 47.789, 130.9873 47.7901, 130.5823 48.7297, 129.3978 49.4406, 127.6574 49.7603, 125.9463 52.7928, 125.0682 53.161, 123.5715 53.4588, 121.0031 53.2514, 120.1771 52.7539, 120.7258 52.5162, 120.7382 51.9641, 119.2794 50.5829, 119.2885 50.1429, 117.8792 49.511, 116.6788 49.8885, 115.4857 49.8052, 114.3625 50.2483, 112.8977 49.5436, 110.662 49.1301, 108.4752 49.2825, 106.8888 50.2743, 103.6765 50.09, 102.2559 50.5106, 102.0652 51.2599, 98.8615 52.0474, 97.8257 51.011, 98.2318 50.4224, 97.2598 49.7261, 94.8159 50.0134, 94.1476 50.4805, 92.2347 50.8022, 88.8056 49.4705, 87.36 49.215, 86.8294 49.8267, 85.5413 49.6929, 83.383 51.0692, 81.946 50.8122, 80.5684 51.3883, 80.0356 50.8648, 77.8009 53.4044, 76.5252 54.177, 76.8911 54.4905, 74.3848 53.5469, 73.4257 53.4898, 73.5085 54.0356, 72.2242 54.3767, 71.1801 54.1333, 70.8653 55.1697, 69.0682 55.3853, 65.1785 54.3542, 61.4366 54.0063, 60.9781 53.665, 61.7 52.98, 60.74 52.72, 60.9273 52.4475, 59.9675 51.9604, 61.588 51.2727, 61.3374 50.7991, 59.9328 50.8422, 59.6423 50.5454, 58.3633 51.0636, 56.778 51.0436, 55.7169 50.6217, 52.3287 51.7187, 50.7666 51.6928, 48.7024 50.6051, 48.5778 49.8748, 47.5495 50.4547, 46.4664 48.3942, 47.3152 47.7159, 48.0573 47.7438, 48.6947 47.0756, 48.5933 46.561, 49.1012 46.3993, 48.6454 45.8063, 47.6759 45.6415, 46.682 44.6092, 47.5909 43.6602, 47.4925 42.9866, 48.5844 41.8089, 47.8157 41.1514, 45.7764 42.0924, 45.4703 42.5028, 43.9312 42.555, 42.3944 43.2203, 39.955 43.435, 36.6755 45.2447, 37.4032 45.4045, 38.233 46.2409, 37.6737 46.6366, 39.1477 47.0448, 39.1212 47.2634, 38.2235 47.1022, 38.2551 47.5464, 39.7383 47.8989, 39.6747 48.7838, 40.069 49.6011, 38.0106 49.9157, 37.3935 50.384, 36.6262 50.2256, 35.3561 50.5772, 35.0222 51.2076, 34.2248 51.256, 34.3917 51.7689, 33.7527 52.3351, 31.786 52.1017, 31.3052 53.074, 32.6936 53.3514, 31.7313 53.794, 30.7575 54.8118, 30.8739 55.551, 29.3716 55.6701, 28.1767 56.1691, 27.77 57.2443, 27.2882 57.4745, 27.7167 57.7919, 27.4202 58.7246, 28.1317 59.3008, 27.9811 59.4754, 29.1177 60.0281, 28.07 60.5035, 31.14 62.3577, 31.5161 62.8677, 30.0359 63.5528, 30.4447 64.2045, 29.5444 64.9487, 30.2177 65.806, 29.0546 66.9443, 29.9774 67.6983, 28.4459 68.3646, 28.5919 69.0648, 32.1327 69.906, 33.7755 69.3014, 36.514 69.0634, 41.0599 67.4571, 41.126 66.7916, 38.383 65.9995, 33.9187 66.7596, 33.1844 66.6325, 34.8148 65.9002, 34.9439 64.4144, 37.0127 63.8498, 37.142 64.3347, 36.5396 64.7645, 37.176 65.1432, 39.5935 64.5208, 40.4356 64.7645, 39.7626 65.4968, 42.0931 66.4762, 43.9498 66.0691, 44.5323 66.7563, 43.6984 67.3525, 44.188 67.9505, 43.4528 68.5708, 46.25 68.25, 46.8213 67.69, 45.5552 67.5665, 45.562 67.0101, 46.3492 66.6677, 47.8942 66.8846, 48.1388 67.5224, 53.7174 68.8574, 54.4717 68.8082, 53.4858 68.2013, 54.7263 68.097, 55.4427 68.4387, 57.317 68.4663, 58.802 68.8808, 59.9414 68.2784, 61.0778 68.9407, 60.03 69.52, 60.55 69.85, 63.504 69.5474, 68.5122 68.0923, 69.1807 68.6156, 68.1352 69.3565, 66.9301 69.4546, 67.2598 69.9287, 66.6947 71.029, 68.5401 71.9345, 69.1964 72.8434, 69.94 73.04, 72.5875 72.7763, 72.796 72.2201, 71.8481 71.409, 72.4701 71.0902, 72.7919 70.3911, 72.5647 69.0209, 73.6679 68.4079, 73.2387 67.7404, 71.28 66.32, 72.423 66.1727, 73.921 66.7895, 74.1865 67.2843, 75.052 67.7605, 74.4693 68.329, 74.9358 68.9892, 73.8424 69.0715, 73.6019 69.6276, 74.3998 70.6318, 73.1011 71.4472, 74.8908 72.1212, 74.6593 72.8323, 75.158 72.855, 75.6835 72.3006, 75.289 71.3356, 76.3591 71.1529, 75.9031 71.874, 77.5767 72.2672, 79.652 72.3201, 81.5 71.75, 80.6107 72.5828, 80.5111 73.6482, 86.8223 73.9369, 86.0096 74.4597, 87.1668 75.1164, 92.9006 75.7733, 93.2342 76.0472, 96.6782 75.9155, 98.9225 76.4469, 100.7597 76.4303, 101.9908 77.2875, 104.3516 77.6979, 106.0666 77.3739, 104.705 77.1274, 106.9701 76.9742)), ((180 70.8322, 178.9034 70.7811, 178.7253 71.0988, 180 71.5157, 180 70.8322)), ((150.7317 75.0841, 149.5759 74.6889, 146.1192 75.173, 146.3585 75.4968, 150.7317 75.0841)), ((145.0863 75.5626, 144.3 74.82, 138.9554 74.6115, 136.9744 75.2617, 137.5118 75.9492, 138.8311 76.1368, 145.0863 75.5626)), ((143.648 50.7476, 144.6541 48.9764, 143.1739 49.3066, 142.5587 47.8616, 143.5335 46.8367, 143.5053 46.1379, 142.7477 46.7408, 142.092 45.9668, 141.9044 48.8592, 142.18 50.9523, 141.5941 51.9354, 141.6825 53.302, 142.6069 53.7621, 142.2097 54.2255, 142.6548 54.3659, 143.648 50.7476)), ((143.6039 73.2124, 139.8631 73.3698, 142.0621 73.8576, 143.6039 73.2124)), ((105.0755 78.3069, 99.4381 77.921, 101.2649 79.234, 102.0864 79.3464, 105.3724 78.7133, 105.0755 78.3069)), ((99.9398 78.8809, 94.9726 79.0447, 93.3129 79.4265, 92.5454 80.1438, 91.1811 80.3415, 95.9409 81.2504, 100.1867 79.7801, 99.9398 78.8809)), ((57.5357 70.7205, 53.6774 70.7627, 53.412 71.2067, 51.6019 71.4748, 51.4558 72.0149, 52.4783 72.2294, 52.4442 72.7747, 54.4276 73.6275, 53.5083 73.7498, 55.9025 74.6275, 55.6319 75.0814, 61.17 76.2519, 68.1571 76.9397, 68.8522 76.5448, 61.5835 75.2609, 58.4771 74.3091, 55.4193 72.3713, 55.6228 71.5406, 57.5357 70.7205)), ((51.1362 80.5473, 47.5861 80.0102, 46.5028 80.2472, 47.0725 80.5594, 44.847 80.5898, 48.3185 80.784, 48.5228 80.5146, 50.0398 80.9189, 51.5229 80.6997, 51.1362 80.5473)), ((22.7311 54.3275, 19.6606 54.4261, 19.8885 54.8662, 21.2684 55.1905, 22.7578 54.8566, 22.7311 54.3275)), ((-175.0142 66.5844, -174.3398 66.3356, -174.5718 67.0622, -171.8573 66.9131, -169.8996 65.9772, -170.8911 65.5414, -172.5302 65.4379, -172.555 64.4608, -172.9553 64.2527, -175.9835 64.9229, -176.2072 65.3567, -178.3599 65.3905, -178.9033 65.7404, -178.6861 66.1121, -179.8838 65.8746, -179.4327 65.4041, -180 64.9797, -180 68.9636, -174.9282 67.2059, -175.0142 66.5844)), ((-178.6938 70.893, -180 70.8322, -180 71.5157, -177.5779 71.2695, -178.6938 70.893))), 250,kilometers)"```

**Pull request merged**